cupy/core/_routines_manipulation.pyx

```diff
@@ -130,6 +130,10 @@ cdef ndarray _ndarray_reshape(ndarray self, tuple shape, order):
     if len(shape) == 1 and cpython.PySequence_Check(shape[0]):
         shape = tuple(shape[0])
 
+    if self.size == 0:
```

Comment:
`xp.array([]).reshape((0,))` should success.

Reply:
Good catch!